crowbar_framework/app/models/ceph_service.rb

```diff
@@ -166,9 +166,9 @@ def create_proposal
 
     # Any spare node after allocating mons and osds is fair game
     # to automatically use as the calamari server
-    calamari_nodes = select_nodes_for_role(nodes, "ceph-calamari")
+    calamari_nodes = select_nodes_for_role(nodes - osd_nodes - mon_nodes, "ceph-calamari")
     calamari_nodes.reject! do |n|
```

Comment:
BTW, wouldn't it be simpler to just reject all cluster members before calling "select_nodes_for_role" and "select_node_for_mds_role" ?